src/ray/common/ray_config_def.h

```diff
@@ -441,3 +441,13 @@ RAY_CONFIG(uint32_t, raylet_start_wait_time_s,
                    getenv("RAY_preallocate_plasma_memory") == std::string("1")
                ? 120
                : 10)
+
+/// The scheduler will treat these predefined resource types as unit_instance.
+/// Default predefined_unit_instance_resources is "GPU".
+/// When set it to "CPU,GPU", we will also treat CPU as unit_instance.
+RAY_CONFIG(string_type, predefined_unit_instance_resources, "GPU")
+
+/// The scheduler will treat these custom resource types as unit_instance.
+/// Default custom_unit_instance_resources is empty.
+/// When set it to "FPGA", we will treat FPGA as unit_instance.
+RAY_CONFIG(string_type, custom_unit_instance_resources, "")
```

Comment:
I‘m a bit confused that why we need this field? FPGA can be treated as a `predefined_unit_instance_resources` as well right?
Let us just remove this field?

Reply:
“FPGA” is a custom resource type, not in `enum PredefinedResources`

Reply:
Thanks. I see.

Reply:
Hmm they could still be combined into one flag right? That would probably be less error-prone.

Reply:
@ericl @chenk008 
How about removing `predefined_unit_instance_resources` and put `GPU` into `custom_unit_instance_resources` by default?

Reply:
And also rename `custom_unit_instance_resources` to `unit_instance_resources`.

Reply:
@ericl @jovany-wang I think it is relatively clear to separate the two flag. The CPU/MEMORY/GPU/OBJECT_MEMORY resource information is detected by ray without any additional config, and the custom_resources should be added into "--resources".